span-normalizer/span-normalizer/src/main/java/org/hypertrace/core/spannormalizer/jaeger/JaegerSpanToAvroRawSpanTransformer.java

```diff
@@ -40,19 +40,17 @@ public void init(ProcessorContext context) {
   }
 
   @Override
-  public KeyValue<TraceIdentity, RawSpan> transform(byte[] key, Span value) {
+  public KeyValue<TraceIdentity, RawSpan> transform(byte[] key, PreProcessedSpan preProcessedSpan) {
+    if (null == preProcessedSpan) {
```

Comment:
so nulls are propagated? Is there anyway to filter those or something? Or are all downstream consumers stuck null checking for the rest of the pipeline? Not familiar with kafka streams.

Reply:
we don't need this check, if the `transform` function returns `null` that record will not be emitted in the stream.